frontend/src/metabase/components/FieldValuesWidget/FieldValuesWidget.tsx

```diff
@@ -438,7 +468,20 @@ export function FieldValuesWidgetInner({
     options,
   });
 
-  const parseFreeformValue = (value: string | number) => {
+  const valueForLabel = (label: string | number) => {
+    const option = fieldValues.find(option => {
+      const opt = option[1] ?? option[0]?.toString();
```

Comment:
I have to say it's quite difficult to follow all this code with `[1]` and `[0]`, it'd be great if we could name these everywhere.